src/dryad/format/endf/createReactionType.hpp

```diff
@@ -0,0 +1,106 @@
+#ifndef NJOY_DRYAD_FORMAT_ENDF_CREATEREACTIONTYPE
+#define NJOY_DRYAD_FORMAT_ENDF_CREATEREACTIONTYPE
+
+// system includes
+#include <vector>
+
+// other includes
+#include "tools/Log.hpp"
+#include "dryad/ReactionType.hpp"
+#include "ENDFtk/Material.hpp"
+#include "ENDFtk/tree/Material.hpp"
+
+namespace njoy {
+namespace dryad {
+namespace format {
+namespace endf {
+
+  /**
+   *  @brief Create the reaction type based of the mt number
+   */
+  ReactionType createReactionType( const ENDFtk::tree::Material& material, int mt ) {
+
+    constexpr std::array< int, 443 > primary = {   2,   5,  11, 17,  19,  20,  21,  22,  23,  24,  25,
```

Comment:
This probably would make more sense as a `std::unordered_set`.  Then you wouldn't have to do a weird binary search.